src/rendervulkan.cpp

```diff
@@ -759,92 +759,101 @@ CVulkanTexture::~CVulkanTexture( void )
 	m_bInitialized = false;
 }
 
-bool vulkan_init_formats()
+bool vulkan_init_format(VkFormat format, uint32_t drmFormat)
 {
-	for ( size_t i = 0; s_DRMVKFormatTable[i].DRMFormat != DRM_FORMAT_INVALID; i++ )
+	// First, check whether the Vulkan format is supported
+	VkPhysicalDeviceImageFormatInfo2 imageFormatInfo = {};
+	imageFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2;
+	imageFormatInfo.format = format;
+	imageFormatInfo.type = VK_IMAGE_TYPE_2D;
+	imageFormatInfo.tiling = VK_IMAGE_TILING_OPTIMAL;
+	imageFormatInfo.usage = VK_IMAGE_USAGE_SAMPLED_BIT;
+	imageFormatInfo.flags = 0;
+	VkImageFormatProperties2 imageFormatProps = {};
+	imageFormatProps.sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2;
+	VkResult res = vkGetPhysicalDeviceImageFormatProperties2( physicalDevice, &imageFormatInfo, &imageFormatProps );
+	if ( res == VK_ERROR_FORMAT_NOT_SUPPORTED )
 	{
-		VkFormat format = s_DRMVKFormatTable[i].vkFormat;
-		uint32_t drmFormat = s_DRMVKFormatTable[i].DRMFormat;
+		return false;
+	}
+	else if ( res != VK_SUCCESS )
+	{
+		vk_errorf( res, "vkGetPhysicalDeviceImageFormatProperties2 failed for DRM format 0x%" PRIX32, drmFormat );
+		return false;
+	}
 
-		// First, check whether the Vulkan format is supported
-		VkPhysicalDeviceImageFormatInfo2 imageFormatInfo = {};
-		imageFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2;
-		imageFormatInfo.format = format;
-		imageFormatInfo.type = VK_IMAGE_TYPE_2D;
-		imageFormatInfo.tiling = VK_IMAGE_TILING_OPTIMAL;
-		imageFormatInfo.usage = VK_IMAGE_USAGE_SAMPLED_BIT;
-		imageFormatInfo.flags = 0;
-		VkImageFormatProperties2 imageFormatProps = {};
-		imageFormatProps.sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2;
-		VkResult res = vkGetPhysicalDeviceImageFormatProperties2( physicalDevice, &imageFormatInfo, &imageFormatProps );
-		if ( res == VK_ERROR_FORMAT_NOT_SUPPORTED )
-		{
-			continue;
-		}
-		else if ( res != VK_SUCCESS )
+	sampledShmFormats.push_back( drmFormat );
```

Comment:
We get duplicates here (one for srgb, one for unorm), does that matter?

Reply:
It shouldn't, but thanks for pointing it out. I will fix it when I wake up... Same for sampledDRMFormats